snuba/web/rpc/common/aggregation.py

```diff
@@ -0,0 +1,280 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from sentry_protos.snuba.v1.trace_item_attribute_pb2 import (
+    AttributeAggregation,
+    ExtrapolationMode,
+    Function,
+)
+
+from snuba.query.dsl import CurriedFunctions as cf
+from snuba.query.dsl import Functions as f
+from snuba.query.dsl import column, literal
+from snuba.query.expressions import (
+    CurriedFunctionCall,
+    Expression,
+    FunctionCall,
+    SubscriptableReference,
+)
+from snuba.web.rpc.common.common import attribute_key_to_expression
+from snuba.web.rpc.common.exceptions import BadSnubaRPCRequestException
+
+sampling_weight_column = column("sampling_weight")
+
+# Z value for 95% confidence interval is 1.96 which comes from the normal distribution z score.
+z_value = 1.96
+
+CUSTOM_COLUMN_PREFIX = "__snuba_custom_column__"
+
+
+@dataclass(frozen=True)
+class CustomColumnInformation:
+    """
+    In order to support extrapolation, we need to be able to create a new column in clickhouse that computes some value, potentially based on some existing column.
+    This class holds the information needed to generate alias for the column so we can know what the column represents when getting results.
+    """
+
+    column_type: str
```

Comment:
what are possible column types?

Reply:
these can be anything, column_type is probably a bad name, I'll change it to something like `custom_column_name`